crates/anvil/src/config.rs

```diff
@@ -997,9 +1037,54 @@ impl NodeConfig {
                 .expect("Failed to establish provider to fork url"),
         );
 
-        let (fork_block_number, fork_chain_id) = if let Some(fork_block_number) =
-            self.fork_block_number
+        let (fork_block_number, fork_chain_id, replay_transactions) = if let Some(fork_choice) =
+            &self.fork_choice
         {
+            let (fork_block_number, replay_transactions) = match fork_choice {
+                ForkChoice::Block(block_number) => (block_number.to_owned(), Vec::new()),
+                ForkChoice::Transaction(transaction_hash) => {
+                    // Determine the block that this transaction was mined in
+                    let transaction = provider
+                        .get_transaction_by_hash(transaction_hash.0.into())
+                        .await
+                        .expect("Failed to get fork transaction")
+                        .unwrap();
+                    let transaction_block_number = transaction.block_number.unwrap();
+
+                    // Record all transactions preceding the specified transaction so as to
+                    // allow replay of them
+                    let transaction_block = provider
+                        .get_block_by_number(transaction_block_number.into(), false)
```

Comment:
here you can request the full block so we can get all the txs with once all
```suggestion
                        .get_block_by_number(transaction_block_number.into(), true)
```